napari/_qt/qt_main_window.py

```diff
@@ -814,8 +814,19 @@ def resize(self, width, height):
         """
         self._qt_window.resize(width, height)
 
-    def show(self):
-        """Resize, show, and bring forward the window."""
+    def show(self, run=False):
+        """Resize, show, and bring forward the window.
+
+        Parameters
+        ----------
+        run : bool, optional
```

Comment:
It's not quite clear to me what `show` with `run=False` actually means, i.e. what is the behaviour of showing without the event loop? Shouldn't there be no kwarg and if someone calls show we start the event loop with `run` if it isn't already started? I might be missing something basic here, so apologies if so

Reply:
i think what probably leads to the confusion here is the fact that in napari, [we actually call `show()`](https://github.com/napari/napari/blob/cc1503acb16a88f779bd22ac3f877e99564b617e/napari/_qt/qt_main_window.py#L158-L159) in the `Window.__init__` and the default is for ``show=True``.  In Qt, widgets are _not_ shown by default.  so to actually see a widget, you do 3 things:
1. create a widget  (`viewer = Viewer()`)
2. show the widget  (this is done implicitly in `Window.__init__` by default)
3. start the event loop  (napari.run())

I get that the our implicit show makes this a little confusing though...  it feels like you're calling show just to "run".  preference?

Reply:
Ok, this makes sense. I think in the context of a napari user and the method on the viewer/ window if you've decided to call `show` than you've also (implicitly?) decided to call `run`, and that we should drop the `run` kwarg and just start the loop if needed. Calling `show` then means you'll always see something.

If you want to remain headless then you shouldn't call `viewer.show`. Can you see any downsides of this @tlambert03? We can ask @jni for a third opinion here too.

You'd then just be able to do

```python
napari.view_path("/some/file.tif").show()
```

and see something, which I think @jni will like

Reply:
> which I think @jni will like

He does! I agree very much that `viewer.show()` should be equivalent to `napari.run()` — especially since all the new run machinery should prevent multiple event loops from happening anyway, right?